main/src/cgeo/geocaching/filters/gui/LogicalFilterViewHolder.java

```diff
@@ -0,0 +1,62 @@
+package cgeo.geocaching.filters.gui;
+
+import cgeo.geocaching.R;
+import cgeo.geocaching.filters.core.GeocacheFilter;
+import cgeo.geocaching.filters.core.GeocacheFilterContext;
+import cgeo.geocaching.filters.core.LogicalGeocacheFilter;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import com.google.android.material.button.MaterialButton;
+import org.apache.commons.lang3.StringUtils;
+
+public class LogicalFilterViewHolder extends BaseFilterViewHolder<LogicalGeocacheFilter> {
+    LogicalGeocacheFilter filter = null;
+    TextView textView = null;
+
+    @Override
+    public View createView() {
+        final LinearLayout ll = new LinearLayout(getActivity());
+        ll.setOrientation(LinearLayout.VERTICAL);
+
+        final LayoutInflater inflater = LayoutInflater.from(getActivity());
+
+        final View infoLine = inflater.inflate(R.layout.checkbox_item, ll, false);
+        ((ImageView) infoLine.findViewById(R.id.item_icon)).setImageResource(R.drawable.ic_menu_filter);
+        infoLine.findViewById(R.id.item_checkbox).setVisibility(View.GONE);
+        textView = infoLine.findViewById(R.id.item_text);
+        ll.addView(infoLine);
+
+        final MaterialButton modify = (MaterialButton) inflater.inflate(R.layout.button_view, ll, false);
+        modify.setText(R.string.cache_filter_nested_filter_modify);
+        modify.setOnClickListener(v -> startSubFilterActivity());
+        ll.addView(modify);
+
+        // restore state
+        setViewFromFilter(createFilterFromView());
+
+        return ll;
+    }
+
+    @Override
+    public void setViewFromFilter(final LogicalGeocacheFilter filter) {
+        this.filter = filter != null ? filter : createFilter();
+        final String text = this.filter.toUserDisplayableString(0);
+        textView.setText(StringUtils.isNotBlank(text) ? text : getActivity().getString(R.string.cache_filter_nested_filter_empty_filterconfic));
+    }
+
+    @Override
+    public LogicalGeocacheFilter createFilterFromView() {
+        return filter;
+    }
+
+    private void startSubFilterActivity() {
+        final GeocacheFilterContext filterContext = new GeocacheFilterContext(GeocacheFilterContext.FilterType.NESTED);
+        filterContext.set(GeocacheFilter.create(null, true, false, filter));
+        ((GeocacheFilterActivity) getActivity()).selectNestedFilter(filterContext, newContext -> setViewFromFilter((LogicalGeocacheFilter) newContext.get().getTree()));
```

Comment:
In my experience, working with lambdas attached to activities leads to uncercoverable results when the activity issuing the call is destroyed while in subactivity (lambdas can't be saved/restored unfortunately; also lambdas are connected to a concrete activity class instance, but this instance is gone of course after destroy/recreate). 

Does your code work also when you are in a nested filter view and then
* the "main" filter activity is destroyed? (I think this can be simulated somehow in developer settings that only one activity remains)
* you change configuration (e.g. by rotating your phone from portrait to landscape)
...and then go bac to main activity?

Reply:
Indeed, the lambda gets lost while rotating the device: `java.lang.NullPointerException: Attempt to invoke interface method 'void androidx.core.util.Consumer.accept(java.lang.Object)' on a null object reference`

Do you have any recommendation how something like this could be implemented in a better way?

Reply:
Unfortunately the only thing I found that works is: use a solution without any lambdas or listeners/callbacks. This makes the code ugly sometimes i am afraid (I had this problem massively with the ImageActivityHelper class).

In your case, I assume you strictly need to fall back to a communication via extra-parameters when opening a "nested filter" activity from the "main" activity, and via "onActivityResult" Intent data to transport something back from nested activity to "main" activity. In your case I assume it should be enough to pass the subfilter's filter config to the main activity, and this main activity can then build it in.

Reply:
>In your case I assume it should be enough to pass the subfilter's filter config to the main activity, and this main activity can then build it in.

Had thought about it as well. However, what happens if we have multiple nested filters inside the global one? Probably it somehow has to keep track of the position where to insert the updated nested filter... I will try my best. :)

Reply:
I think I've finally come up with a working solution. Wouldn't mind another review ;-)